e2e/specs/age-based-vitals.spec.ts

```diff
@@ -0,0 +1,212 @@
+        await vitalsPage.page.getByRole('spinbutton', { name: /temperature/i }).fill(group.criticalVitals.temp);
+      });
+
+      await test.step(`And I enter the systolic blood pressure: ${group.criticalVitals.systolic}`, async () => {
+        await vitalsPage.page.getByRole('spinbutton', { name: /systolic/i }).fill(group.criticalVitals.systolic);
+      });
+
+      await test.step(`And I enter the diastolic blood pressure: ${group.criticalVitals.diastolic}`, async () => {
+        await vitalsPage.page.getByRole('spinbutton', { name: /diastolic/i }).fill(group.criticalVitals.diastolic);
+      });
+
+      await test.step(`And I enter the respiration rate: ${group.criticalVitals.respiration}`, async () => {
+        await vitalsPage.page
+          .getByRole('spinbutton', { name: /respiration rate/i })
+          .fill(group.criticalVitals.respiration);
+      });
+
+      await test.step('And I click on the `Save and close` button to save the vitals', async () => {
+        await vitalsPage.page.getByRole('button', { name: /save and close/i }).click();
+      });
+
+      await test.step('Then I should see a success notification confirming vitals were saved', async () => {
+        await expect(vitalsPage.page.getByText(/vitals and biometrics saved/i)).toBeVisible();
+      });
+
+      await test.step('And I should see the vitals table headers', async () => {
+        await expect(headerRow).toContainText(/temp/i);
+        await expect(headerRow).toContainText(/bp/i);
+        await expect(headerRow).toContainText(/r. rate/i);
+      });
+
+      await test.step(`And I should see the temperature value of ${group.criticalVitals.temp} in the vitals table`, async () => {
+        await expect(dataRow).toContainText(group.criticalVitals.temp);
+      });
+
+      await test.step(`And I should see the blood pressure value of ${group.criticalVitals.systolic}/${group.criticalVitals.diastolic} in the vitals table`, async () => {
+        await expect(dataRow).toContainText(`${group.criticalVitals.systolic} / ${group.criticalVitals.diastolic}`);
+      });
+
+      await test.step(`And I should see the respiration rate of ${group.criticalVitals.respiration} in the vitals table`, async () => {
+        await expect(dataRow).toContainText(group.criticalVitals.respiration);
+      });
+
+      await test.step('And the temperature cell should have warning styling', async () => {
+        const criticalCell = vitalsPage.page.getByRole('cell', { name: group.criticalVitals.temp });
+        const backgroundColor = await criticalCell.evaluate((el) => window.getComputedStyle(el).backgroundColor);
+        expect(backgroundColor).toBe('rgb(255, 242, 232)');
+
+        const afterContent = await criticalCell.evaluate((el) => {
+          const after = window.getComputedStyle(el, '::after');
+          return after.content;
+        });
+        expect(afterContent).toBe('" ↑"');
+      });
+    });
+  });
+});
+
+test.afterEach(async ({ api }) => {
```

Comment:
Why is this `afterEach()` not just part of the per-each group. That way we don't need some sort of global visit and patient tracker. See [the Playwright docs for an example](https://playwright.dev/docs/test-parameterize#before-and-after-hooks).